librz/io/p/io_ihex.c

```diff
@@ -68,14 +68,14 @@ static int fw04b(FILE *fd, ut16 eaddr) {
 
 //write contiguous block of data to file; ret 0 if ok
 //max 65535 bytes; assumes a 04 rec was written before
-static int fwblock(FILE *fd, ut8 *b, ut32 start_addr, ut16 size) {
+static int fwblock(FILE *fd, ut8 *b, ut32 start_addr, ut32 size) {
 	ut8 cks;
 	char linebuf[80];
 	ut16 last_addr;
 	int j;
 	ut32 i; //has to be bigger than size !
 
-	if (size < 1 || !fd || !b) {
+	if (size < 1 || size > 0x10000 || !fd || !b) {
```

Comment:
Unnecessary for this PR, but maybe makes sense to extract this and 0xffff as this constant -1

Reply:
Ok, forget that.